addon/-debug/helpers/subclass-of.js

```diff
@@ -1,10 +1,9 @@
 import { assert } from '@ember/debug';
 import { makeValidator } from '../utils/validators';
-import isClass from '../utils/is-class';
 
 export default function subclassOf(superclass) {
   assert(`The 'subclassOf' helper must receive exactly one class`, arguments.length === 1);
-  assert(`The 'subclassOf' helper must receive a class constructor, received: ${superclass}`, isClass(superclass));
+  assert(`The 'subclassOf' helper must receive a class constructor, received: ${superclass}`, typeof superclass === 'function');
```

Comment:
Because of the type of check we're doing here (traversing up the prototype chain to figure out if it's an subclass), we can be less strict about the class check.